packages/web-app/src/modules/earn-views/components/ShowFolderSection.tsx

```diff
@@ -0,0 +1,47 @@
+import { Component } from 'react'
+import withStyles, { WithStyles } from 'react-jss'
+import { Button, Divider, P, SectionHeader } from '../../../components'
+import { SaladTheme } from '../../../SaladTheme'
+
+const styles = (theme: SaladTheme) => ({
+  container: {
+    flexDirection: 'column',
+    color: theme.lightGreen,
+  },
+  content: {
+    padding: 20,
+    paddingRight: 20,
+    color: theme.lightGreen,
+  },
+})
+
+interface Props extends WithStyles<typeof styles> {
+  openFolder: () => void
+}
+
+class _ShowFolderSection extends Component<Props> {
+  handleOpenFolder = () => {
+    const { openFolder } = this.props
+    openFolder()
```

Comment:
@scottcinatl @jpark799 I see this happening a decent amount... Why? What's wrong with the following?

```suggestion
    this.props.openFolder()
```

Reply:
@seniorquico @jpark799 I was just going to comment on this as well. This is a common pattern that other components in the app followed. But I believe this pattern originated only when a function being passed down to a component was potentially undefined. So it took this form: 

```
handleClick = () => {
    const { openFolder } = this.props
    if (openFolder) {
        openFolder()
    }
}
```
Since the `openFolder()` function is required, we could even just change this to a functional component and pass down the function directly to the click event `<Button onClick={openFolder}>Show Folder</Button>`.

Reply:
As I've been digging through the code base, I've been trying to follow the patterns that were already set. Thank you for adding that context around how it came to be, Scott. 

This could definitely just be a functional component and doing that would make it a little easier to read too imo. I've gone ahead and made the change. Thank you!